src/main/kotlin/no/nav/klage/oppgave/clients/klagefileapi/FileApiClient.kt

```diff
@@ -0,0 +1,80 @@
+package no.nav.klage.oppgave.clients.klagefileapi
+
+import brave.Tracer
+import no.nav.klage.oppgave.util.TokenUtil
+import no.nav.klage.oppgave.util.getLogger
+import org.springframework.http.HttpHeaders
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+
+@Component
+class FileApiClient(
+    private val fileWebClient: WebClient,
+    private val tokenUtil: TokenUtil,
+    private val tracer: Tracer
+) {
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+    }
+
+    fun getDocument(id: String): ByteArray {
+        logger.debug("Fetching document with id {}", id)
+
+        return this.fileWebClient.get()
+            .uri { it.path("/document/{id}").build(id) }
+            .header("Nav-Consumer-Token", "Bearer ${tokenUtil.getStsSystembrukerToken()}")
```

Comment:
Hva brukes denne til tro? Er det ikke kun et kall mot vår egen tjeneste?

Reply:
Jeg har prøvd å bruke OBO-flyten her, i stedet for maskin-til-maskin-flyt i autentiseringen. Jeg kan gjøre et forsøk i miljøet uten denne, er interessant å se.